implementations/rust/ockam/ockam_command/src/identity/delete.rs

```diff
@@ -0,0 +1,39 @@
+use crate::util::{exitcode, node_rpc};
+use crate::CommandGlobalOpts;
+use clap::Args;
+use anyhow::anyhow;
+use ockam::Context;
+
+#[derive(Clone, Debug, Args)]
+pub struct DeleteCommand {
+    name: String,
+}
+
+impl DeleteCommand {
+    pub fn run(self, options: CommandGlobalOpts) {
+        node_rpc(run_impl, (options, self))
+    }
+}
+
+async fn run_impl(
+    ctx: Context,
+    (opts, cmd): (CommandGlobalOpts, DeleteCommand),
+) -> crate::Result<()> {
+    let identity_state = opts.state.identities.get(&cmd.name)?;
+    let identity = identity_state.config.identifier.to_string();
+    for node in opts.state.nodes.list()? {
+        let node_identity = node.config.identity(&ctx).await?;
+        let node_identity = node_identity.identifier().to_string();
+        if node_identity == identity {
+            return Err(crate::Error::new(
+                exitcode::USAGE,
+                anyhow!("Cannot delete identity that is being used"),
+            ));
+        }
+    }
+    let default = opts.state.identities.default()?;
+    let delete_symlink = default.config.identifier.to_string() == identity;
```

Comment:
This logic should be inside the `CliState` because we don't need any information from outside to figure out if we need to delete the symlink.